File: fe-nextjs/types/folder.ts
import { ChatSession } from "./chatsession";

export interface MenuItem {
    key: string;
    icon?: React.ReactNode;
    label: string;
    isNew?: boolean;
    isFolder?: boolean;
    isSubItem?: boolean;
    isExpanded?: boolean;
    itemCount?: number;
    parentFolder?: string;
}


// export type Folder = {
//     id: string;
//     title: string;
//     chatSessions: ChatSession[];
//     createdDate: string;
//     updatedDate: string;
// }

// export type Folder = {
//     id: string;
//     title: string;
//     description?: string;
//     patientProfileId: string | null;
//     chatSessionIds?: string[];
//     createdBy: string;
//     isDeleted: boolean;
//     createdDate?: string;
//     updatedDate?: string;
// }

export type Folder = {
    id: string;
    title: string;
    description?: string;
    patientProfileId: string | null;
    chatSessionIds?: string[];
    chatSessionsInfo?: ChatSessionInfo[];
    createdBy: string;
    isDeleted: boolean;
    createdDate?: Date;
    updatedDate?: Date;
}

export type ChatSessionInfo = {
    id: string;
    title: string;
    isDeleted: boolean;
    createdDate?: Date;
    updatedDate?: Date;
}

File: servers/DoctorService/Models/Folder.cs
using System.ComponentModel.DataAnnotations;

namespace DoctorService.Models
{
    public class Folder
    {
        [Key]
        public string Id { get; set; } = string.Empty;
        
        [Required]
        public string Title { get; set; } = string.Empty;
        
        public string? Description { get; set; }
        
        public string? PatientProfileId { get; set; } // patient profile id
        
        public List<string>? ChatSessionIds { get; set; }
        
        [Required]
        public string CreatedBy { get; set; } = string.Empty; // user id
        
        [Required]
        public bool IsDeleted { get; set; }
        
        public DateTime? CreatedDate { get; set; }
        
        public DateTime? UpdatedDate { get; set; }
    }
}

